generator/src/pkg/generate/generate.go

```diff
@@ -194,23 +225,33 @@ func Create(m map[string]map[string][]string, placement Placement, readinessProb
 				manifests = append(manifests, string(yamlDoc))
 				return nil
 			}
-			configData = ConfigData{Hop: ChainTypes{Chain1: m["1"][serv], Chain2: m["2"][serv], Chain3: m["3"][serv], Chain4: m["4"][serv], Chain5: m["5"][serv],
-				Chain6: m["6"][serv], Chain7: m["7"][serv], Chain8: m["8"][serv]}, Hostname: serv}
-			d, _ := json.Marshal(configData)
-			data := fmt.Sprintf("%s", string(d))
-
-			config = s.CreateConfig("config-"+serv, "config-"+serv, c_id, namespace, data)
-			appendManifest(config)
-			deployment = s.CreateDeployment(serv, serv, c_id, replicaNumber, serv, c_id, namespace,
-			    defaultPort, redisDefaultPort, fortioDefaultPort, imageName, imageURL, redisImageName,
-				redisImageURL, workerImageName, workerImageURL, fortioImageName, fortioImageUrl, volumePath, volumeName, "config-"+serv, readinessProbe)
-			appendManifest(deployment)
+			configmap = s.CreateConfig("config-"+serv, "config-"+serv, c_id, namespace, string(serv_ep_json))
+			appendManifest(configmap)
+			if nodeAffinity == "" {
+				deployment := s.CreateDeployment(serv, serv, c_id, replicaNumber, directory, c_id, namespace,
+					defaultPort, redisDefaultPort, fortioDefaultPort, imageName, imageURL, redisImageName,
+					redisImageURL, workerImageName, workerImageURL, fortioImageName, fortioImageUrl, volumePath, volumeName, "config-"+serv, readinessProbe,
+					resources.Requests.Cpu, resources.Requests.Memory, resources.Limits.Cpu, resources.Limits.Memory)
+
+				appendManifest(deployment)
+			} else {
+				deployment := s.CreateDeploymentWithAffinity(serv, serv, c_id, replicaNumber, directory, c_id, namespace,
+					defaultPort, redisDefaultPort, fortioDefaultPort, imageName, imageURL, redisImageName,
+					redisImageURL, workerImageName, workerImageURL, fortioImageName, fortioImageUrl, volumePath, volumeName, "config-"+serv, readinessProbe,
+					resources.Requests.Cpu, resources.Requests.Memory, resources.Limits.Cpu, resources.Limits.Memory, nodeAffinity)
+				appendManifest(deployment)
+			}
+
 			service = s.CreateService(serv, serv, protocol, fortioProtocol, fortioUiProtocol, uri, fortioUri, fortioUiUri, c_id, namespace, defaultExtPort, defaultPort, fortioDefaultPort, fortioUiDefaultPort)
 			appendManifest(service)
 
 			yamlDocString := strings.Join(manifests, "---\n")
-			ioutil.WriteFile(manifestFilePath, []byte(yamlDocString), 0644)
+			err := ioutil.WriteFile(manifestFilePath, []byte(yamlDocString), 0644)
+			if err != nil {
+				fmt.Print(err)
+				return
 			}
-	}
 
+		}
+	}
 }
```

Comment:
Is there any example of generated k8s yaml file you have tested with that could be attache

Reply:
The output k8s yaml file is as below:
```
apiVersion: v1
kind: ConfigMap
metadata:
    name: config-service_1
    labels:
        name: config-service_1
        version: cluster_1
    namespace: ns_1
data:
    conf.json: '[{"name":"/end_1","cpuConsumption":0,"networkConsumption":0,"memoryConsumption":0,"calledServices":[{"cluster":"cluster_1","service":"service_2","endpoint":"/end_2","protocol":"","traffic_forward_ratio":1,"requests":"asynchronous"}],"requests":""}]'
---
apiVersion: apps/v1
kind: Deployment
metadata:
    name: service_1
    namespace: ns_1
    labels:
        version: cluster_1
spec:
    selector:
        matchLabels:
            app: service_1
            version: cluster_1
    replicas: 1
    template:
        metadata:
            labels:
                app: cluster_1
                version: cluster_1
        spec:
            nodeName: node_1
            containers:
                - name: app
                  image: app-demo:latest
                  imagePullPolicy: Never
                  ports:
                    - containerPort: 5000
                  volumeMounts:
                    - mountPath: /usr/src/app/config
                      name: config-data-volume
                  readinessProbe:
                    httpGet:
                        path: /
                        port: 5000
                    initialDelaySeconds: 10
                    periodSeconds: 1
                  resources:
                    limits:
                        cpu: "1000"
                        memory: "1024"
                    requests:
                        cpu: "500"
                        memory: "512"
            volumes:
                - name: config-data-volume
                  configMap:
                    name: config-service_1
---
apiVersion: v1
kind: Service
metadata:
    name: service_1
    namespace: ns_1
    labels:
        version: cluster_1
    annotations:
        http: /
spec:
    selector:
        app: service_1
    ports:
        - name: http
          port: 80
          targetPort: 5000
```